packages/ses/src/get-anonymous-intrinsics.js

```diff
@@ -135,6 +118,33 @@ export const getAnonymousIntrinsics = () => {
     '%InertCompartment%': InertCompartment,
   };
 
+  try {
+    // 25.3.1 The AsyncGeneratorFunction Constructor
+
+    // eslint-disable-next-line no-empty-function
+    const AsyncGeneratorFunctionInstance = new FERAL_FUNCTION(
+      'return (async function* AsyncGeneratorFunctionInstance() {})',
+    )();
+    const AsyncGeneratorFunction = getConstructorOf(
+      AsyncGeneratorFunctionInstance,
+    );
+
+    // 25.3.2.2 AsyncGeneratorFunction.prototype
+    const AsyncGenerator = AsyncGeneratorFunction.prototype;
+    // 25.5.1 Properties of the AsyncGenerator Prototype Object
+    const AsyncGeneratorPrototype = AsyncGenerator.prototype;
+    const AsyncIteratorPrototype = getPrototypeOf(AsyncGeneratorPrototype);
+
+    assign(intrinsics, {
+      '%AsyncGenerator%': AsyncGenerator,
+      '%InertAsyncGeneratorFunction%': AsyncGeneratorFunction,
+      '%AsyncGeneratorPrototype%': AsyncGeneratorPrototype,
+      '%AsyncIteratorPrototype%': AsyncIteratorPrototype,
+    });
+  } catch {
+    // Silently swallow Hermes SyntaxError `async generators are unsupported` at runtime
+  }
```

Comment:
```suggestion
  // Test for support async generator function syntax.
  let AsyncGeneratorFunctionInstance;
  try {
    // eslint-disable-next-line no-empty-function
    AsyncGeneratorFunctionInstance = new FERAL_FUNCTION(
      'return (async function* AsyncGeneratorFunctionInstance() {})',
    )();
  } catch {
    // Silently swallow Hermes SyntaxError `async generators are unsupported` at runtime
  }
  
  if (AsyncGeneratorFunctionInstance !== undefined) {  
    // 25.3.1 The AsyncGeneratorFunction Constructor

    const AsyncGeneratorFunction = getConstructorOf(
      AsyncGeneratorFunctionInstance,
    );

    // 25.3.2.2 AsyncGeneratorFunction.prototype
    const AsyncGenerator = AsyncGeneratorFunction.prototype;
    // 25.5.1 Properties of the AsyncGenerator Prototype Object
    const AsyncGeneratorPrototype = AsyncGenerator.prototype;
    const AsyncIteratorPrototype = getPrototypeOf(AsyncGeneratorPrototype);

    assign(intrinsics, {
      '%AsyncGenerator%': AsyncGenerator,
      '%InertAsyncGeneratorFunction%': AsyncGeneratorFunction,
      '%AsyncGeneratorPrototype%': AsyncGeneratorPrototype,
      '%AsyncIteratorPrototype%': AsyncIteratorPrototype,
    });
  }
```

Reply:
@kriskowal , my comment above

> This should also check that the error is a SyntaxError, and if not, rethrow it stopping ses initialization.
>
> Never assume you know what a thrown error is without looking, or you'll mask other problems.

applies to your suggested replacement as well.